### ISSUE TITLE
Unique ID enforcement?

### ISSUE THREAD

**Issue by artificial-aidan** [morintd/prismock#685]
I'm using `@@id` to set a field as the ID, but prismock lets me create multiple objects with the same ID. Is this expected? I couldn't find anywhere in the code that would enforce that.

**Comment by morintd** [comment]
Hello @artificial-aidan , there is indeed no mechanism that prevents you from using the same ID multiple times. More generally, errors are not yet implemented.

Now that Prismock is stable enough IMO, I'm starting to think about introducing those mechanisms. I might do so after the current issues get fixed.

**Comment by charlesilieff** [comment]
I am interested in the Prismock project, and I noticed the recent issue raised by @artificial-aidan. I was wondering if you have plans to introduce such a mechanism soon.

If you do not have plans to implement this feature in the near future, I would be interested in contributing to the project by implementing it myself. I believe that enforcing ID uniqueness is important for ensuring data integrity, and I would be excited to help improve Prismock in this way.

**Comment by sledorze** [comment]
@morintd Hello, interested also in this library, if you may give some directions, maybe we can help?

**Comment by morintd** [comment]
Hello @artificial-aidan , @charlesilieff , @sledorze ,

I just realised there was activity on this thread for @@id, which has been implemented by @bertrand-sifre in https://github.com/morintd/prismock/pull/894.

Sorry for the wait, it should be available soon.

**Comment by thw0rted** [comment]
Would it make sense to implement `@unique` at the same time?  I'm not sure how the behavior differs between those, under the hood.

**Comment by thw0rted** [comment]
After #1105 was merged, I was expecting `baz.create({data: {fooId, barId}})` to fail when creating a record that links to the ID of a "foo" or "bar" that doesn't exist.  I see now that the PR didn't update the "create" delegate so I'm guessing more changes will be needed to support that kind of foreign key check.  Does that still fit under the spirit of this ticket, or is a new ticket required?
